elf_reader.go

```diff
@@ -585,6 +587,40 @@ func (ec *elfCode) relocateInstruction(ins *asm.Instruction, rel elf.Symbol) err
 			return fmt.Errorf("asm relocation: %s: unsupported type %s", name, typ)
 		}
 
+		// If a Call instruction is found and the datasec has a btf.Func with a Name
+		// that matches the symbol name we mark the instruction as a call to a kfunc.
+		if ins.OpCode.JumpOp() == asm.Call {
```

Comment:
Maybe turn this into a switch together with the `ins.OpCode.IsDWordLoad()` check below?

I think it'd also be good to label the break statements here to be more explicit.

Reply:
Refactoring some parts made lots of breaks obsolete.